pandas/core/indexes/range.py

```diff
@@ -596,6 +597,9 @@ def _evaluate_numeric_binop(self, other):
                     # GH#19333 is_integer evaluated True on timedelta64,
                     # so we need to catch these explicitly
                     return op(self._int64index, other)
+                elif is_timedelta64_dtype(other):
```

Comment:
doesn't the fall thru raise TypeError?

Reply:
I don’t understand the question

Reply:
your comment is at odds with the checking i think. pls revise comments and/or checks.

Reply:
The comment below this line `# must be an np.ndarray; GH#22390` along with the check above `elif is_timedelta64_dtype(other)` is just saying this is an `ndarray["timedelta64['ns']"]`.  I don't think these are at odds.